lib/cook_book.rb

```diff
@@ -0,0 +1,33 @@
+class CookBook
+  attr_reader :recipes, :date
+
+  def initialize
+    @recipes = []
+    @date = format_time(Time.new)
+  end
+
+  def add_recipe(recipe)
+    @recipes << recipe
+  end
+
+  def ingredients
+    @recipes.flat_map do |recipe|
+      recipe.ingredients.map do |ingredient|
+        ingredient.name
+      end
+    end.uniq
+  end
+
+  def highest_calorie_meal
+    @recipes.max_by do |recipe|
+      recipe.total_calories.to_i
+    end
+  end
+
+  def format_time(time)
+    day = time.day.to_s.rjust(2, '0')
```

Comment:
I love your creativity here. Look up the strftime method though. It will be helpful with this situation :)